[BUG] OCR_USER_ARGS not used when forcing ocr

### Description

During a fallback ocr attempt with force_ocr: true, any provided OCR_USER_ARGS are ignored.

### Steps to reproduce

1. Have a file with both a digital signature and a user fillable form.
2. Have `"invalidate_digital_signatures": true` set in `PAPERLESS_OCR_USER_ARGS` so digital signatures are ignored when consuming.
3. Attempt to consume or to redo ocr on the file.
4. First attempt to ocr will fail because of the form.
5. Second automated attempt with `'force_ocr': true` will also fail because this time `PAPERLESS_OCR_USER_ARGS` will not be used at all and an error will be thrown because of the included signature.

### Webserver logs

```bash
[2023-12-13 21:49:04,119] [DEBUG] [paperless.parsing.tesseract] Calling OCRmyPDF with args: {'input_file': PosixPath('/tmp/paperless/paperless-ngx1fy9lzv9/Consors_Contract_iPhone_15_Pro.pdf'), 'output_file': PosixPath('/tmp/paperless/paperless-jz9xb_sg/archive.pdf'), 'use_threads': True, 'jobs': 4, 'language': 'eng+deu+pol', 'output_type': 'pdfa', 'progress_bar': False, 'color_conversion_strategy': 'RGB', 'redo_ocr': True, 'clean': True, 'rotate_pages': True, 'rotate_pages_threshold': 12.0, 'sidecar': PosixPath('/tmp/paperless/paperless-jz9xb_sg/sidecar.txt'), 'invalidate_digital_signatures': True}

[2023-12-13 21:49:11,642] [WARNING] [paperless.parsing.tesseract] Encountered an error while running OCR: This PDF has a user fillable form. --redo-ocr is not currently possible on such files.. Attempting force OCR to get the text.

[2023-12-13 21:49:11,642] [DEBUG] [paperless.parsing.tesseract] Fallback: Calling OCRmyPDF with args: {'input_file': PosixPath('/tmp/paperless/paperless-ngx1fy9lzv9/Consors_Contract_iPhone_15_Pro.pdf'), 'output_file': PosixPath('/tmp/paperless/paperless-jz9xb_sg/archive-fallback.pdf'), 'use_threads': True, 'jobs': 4, 'language': 'eng+deu+pol', 'output_type': 'pdfa', 'progress_bar': False, 'color_conversion_strategy': 'RGB', 'force_ocr': True, 'clean': True, 'rotate_pages': True, 'rotate_pages_threshold': 12.0, 'sidecar': PosixPath('/tmp/paperless/paperless-jz9xb_sg/sidecar-fallback.txt')}

[2023-12-13 21:49:12,793] [ERROR] [paperless.consumer] Error occurred while consuming document Consors_Contract_iPhone_15_Pro.pdf: DigitalSignatureError: Input PDF has a digital signature. OCR would alter the document,

invalidating the signature.

Traceback (most recent call last):

  File "/usr/src/paperless/src/paperless_tesseract/parsers.py", line 330, in parse

    ocrmypdf.ocr(**args)

  File "/usr/local/lib/python3.11/site-packages/ocrmypdf/api.py", line 375, in ocr

    return run_pipeline(options=options, plugin_manager=plugin_manager)

           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/usr/local/lib/python3.11/site-packages/ocrmypdf/_pipelines/ocr.py", line 225, in run_pipeline

    return _run_pipeline(options, plugin_manager)

           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/usr/local/lib/python3.11/site-packages/ocrmypdf/_pipelines/ocr.py", line 189, in _run_pipeline

    validate_pdfinfo_options(context)

  File "/usr/local/lib/python3.11/site-packages/ocrmypdf/_pipeline.py", line 205, in validate_pdfinfo_options

    raise InputFileError(

ocrmypdf.exceptions.InputFileError: This PDF has a user fillable form. --redo-ocr is not currently possible on such files.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):

  File "/usr/src/paperless/src/paperless_tesseract/parsers.py", line 381, in parse

    ocrmypdf.ocr(**args)

  File "/usr/local/lib/python3.11/site-packages/ocrmypdf/api.py", line 375, in ocr

    return run_pipeline(options=options, plugin_manager=plugin_manager)

           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/usr/local/lib/python3.11/site-packages/ocrmypdf/_pipelines/ocr.py", line 225, in run_pipeline

    return _run_pipeline(options, plugin_manager)

           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/usr/local/lib/python3.11/site-packages/ocrmypdf/_pipelines/ocr.py", line 189, in _run_pipeline

    validate_pdfinfo_options(context)

  File "/usr/local/lib/python3.11/site-packages/ocrmypdf/_pipeline.py", line 202, in validate_pdfinfo_options

    raise DigitalSignatureError()

ocrmypdf.exceptions.DigitalSignatureError: Input PDF has a digital signature. OCR would alter the document,

invalidating the signature.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):

  File "/usr/local/lib/python3.11/site-packages/asgiref/sync.py", line 349, in main_wrap

    raise exc_info[1]

  File "/usr/src/paperless/src/documents/consumer.py", line 446, in try_consume_file

    document_parser.parse(self.path, mime_type, self.filename)

  File "/usr/src/paperless/src/paperless_tesseract/parsers.py", line 393, in parse

    raise ParseError(f"{e.__class__.__name__}: {e!s}") from e

documents.parsers.ParseError: DigitalSignatureError: Input PDF has a digital signature. OCR would alter the document,

invalidating the signature.
```


### Browser logs

_No response_

### Paperless-ngx version

2.1.2

### Host OS

Paperless runs in Docker on QNAP NAS with Intel CPU

### Installation method

Docker - official image

### Browser

_No response_

### Configuration changes

      PAPERLESS_OCR_LANGUAGES: eng deu pol       PAPERLESS_OCR_LANGUAGE: eng+deu+pol       PAPERLESS_OCR_MODE: redo       PAPERLESS_OCR_USER_ARGS: '{"invalidate_digital_signatures": true}'

### Other

The first three lines of the logs show what I assume to be the cause of the error, the argument is not visible in the third line.

--- a/docs/configuration.md
+++ b/docs/configuration.md
@@ -733,7 +733,7 @@ they use underscores instead of dashes.
         Paperless has been tested to work with the OCR options provided
         above. There are many options that are incompatible with each other,
         so specifying invalid options may prevent paperless from consuming
-        any documents.
+        any documents.  Use with caution!
 
     Specify arguments as a JSON dictionary. Keep note of lower case
     booleans and double quoted parameter names and strings. Examples:

--- a/src/paperless_tesseract/parsers.py
+++ b/src/paperless_tesseract/parsers.py
@@ -254,7 +254,7 @@ class RasterisedDocumentParser(DocumentParser):
                     f"Image DPI of {ocrmypdf_args['image_dpi']} is low, OCR may fail",
                 )
 
-        if settings.OCR_USER_ARGS and not safe_fallback:
+        if settings.OCR_USER_ARGS:
             try:
                 user_args = json.loads(settings.OCR_USER_ARGS)
                 ocrmypdf_args = {**ocrmypdf_args, **user_args}

